lib/ansible/plugins/action/win_reboot.py

```diff
@@ -23,201 +22,68 @@ class TimedOutException(Exception):
-
-        test_command = str(self._task.args.get('test_command', self.DEFAULT_TEST_COMMAND))
-        msg = str(self._task.args.get('msg', self.DEFAULT_REBOOT_MESSAGE))
-
-        # Get current uptime
-        try:
-            before_uptime = self.get_system_uptime()
-        except Exception as e:
-            result['failed'] = True
-            result['reboot'] = False
-            result['msg'] = to_native(e)
-            return result
+        remote_command = self.construct_command()
+        reboot_result = self._low_level_execute_command(remote_command, sudoable=self.DEFAULT_SUDOABLE)
 
-        # Initiate reboot
-        display.vvv("rebooting server")
-        (rc, stdout, stderr) = self._connection.exec_command('shutdown /r /t %d /c "%s"' % (pre_reboot_delay, msg))
+        pre_reboot_delay = int(self._task.args.get('pre_reboot_delay', self._task.args.get('pre_reboot_delay_sec', self.DEFAULT_PRE_REBOOT_DELAY)))
 
         # Test for "A system shutdown has already been scheduled. (1190)" and handle it gracefully
-        if rc == 1190 or (rc != 0 and b"(1190)" in stderr):
+        if reboot_result['rc'] == 1190 or (reboot_result['rc'] != 0 and b"(1190)" in reboot_result['stderr']):
             display.warning('A scheduled reboot was pre-empted by Ansible.')
 
             # Try to abort (this may fail if it was already aborted)
-            (rc, stdout1, stderr1) = self._connection.exec_command('shutdown /a')
+            result1 = self._low_level_execute_command('shutdown /a', sudoable=self.DEFAULT_SUDOABLE)
+            # (reboot_result['rc'], stdout1, stderr1) = self._connection.exec_command('shutdown /a')
 
             # Initiate reboot again
-            (rc, stdout2, stderr2) = self._connection.exec_command('shutdown /r /t %d' % pre_reboot_delay)
-            stdout += stdout1 + stdout2
-            stderr += stderr1 + stderr2
+            result2 = self._connection.exec_command('shutdown /r /t %d' % pre_reboot_delay)
+            # (reboot_result['rc'], stdout2, stderr2) = self._connection.exec_command('shutdown /r /t %d' % pre_reboot_delay)
 
-        if rc != 0:
+            stdout = reboot_result['stdout'] + result1['stdout'] + result2['stdout']
+            stderr = reboot_result['stderr'] + result1['stderr'] + result2['stderr']
+
+        result = {}
+        if reboot_result['rc'] != 0:
             result['failed'] = True
             result['rebooted'] = False
-            result['msg'] = "Shutdown command failed, error text was '%s'" % to_native(stderr)
+            result['msg'] = "Shutdown command failed, error was: %s %s" % (to_native(stdout.strip()), to_native('stderr'.strip()))
             return result
 
-        start = datetime.now()
+        result['failed'] = False
+        result['start'] = datetime.utcnow()
+
         # Get the original connection_timeout option var so it can be reset after
-        connection_timeout_orig = None
+        result['connection_timeout_orig'] = None
```

Comment:
You're using the result object to smuggle this around, but it should probably be an instance field instead, since we don't actually want it in the results. Probably the same for the other cases where we're using the result dict as intermediate storage- put them in fields and just construct the result object at the end before it's returned.